daemon/src/maker_inc_connections.rs

```diff
@@ -124,9 +144,22 @@ impl Handler<TakerMessage> for Actor {
 }
 
 #[async_trait]
-impl Handler<NewTakerOnline> for Actor {
-    async fn handle(&mut self, msg: NewTakerOnline, _ctx: &mut Context<Self>) {
-        log_error!(self.handle_new_taker_online(msg));
+impl Handler<ListenerMessage> for Actor {
+    async fn handle(&mut self, msg: ListenerMessage, _ctx: &mut Context<Self>) -> KeepRunning {
+        match msg {
+            ListenerMessage::NewConnection { stream, address } => {
+                self.handle_new_connection(stream, address).await;
+
+                KeepRunning::Yes
+            }
+            ListenerMessage::Error { source } => {
+                tracing::warn!("TCP listener produced an error: {}", source);
+
+                // Maybe we should move the actual listening on the socket into here and restart the
+                // actor upon an error?
```

Comment:
an interesting idea, definitely worth exploring - is there a way of recovering from this otherwise? However, not sure whether we need it for the MVP?

Reply:
> we need it for the MVP?

Probably not but something to keep in mind :)